src/main/java/org/spongepowered/api/command/args/GenericArguments.java

```diff
@@ -929,14 +929,14 @@ public static CommandElement bool(Text key) {
 
         @Override
         protected Iterable<String> getChoices(CommandSource source) {
-            return Arrays.asList(this.type.getEnumConstants()).stream()
-                .map(input -> input == null ? null : input.name())
+            return Arrays.stream(this.type.getEnumConstants())
+                .map(Enum::name)
                 .collect(Collectors.toList());
         }
 
         @Override
         protected Object getValue(String choice) throws IllegalArgumentException {
-            return Enum.valueOf(this.type, choice.toUpperCase());
+            return Enum.valueOf(this.type, choice);
```

Comment:
Should I worry about cases when there are two different values with the same value but different capitilisation, as that is possible with enums. Obviously this is terrible practice, but with the implementation you suggested, there would be no way to differentiate between them.